### PR TITLE
Parser for RecycleBin $I files and RecycleBin Artefact

### PR DESCRIPTION
**Pull request by svch0stz**
Will parse the $I files in the $Recycle.Bin. Parser was generated with binparsergen tool.

Sample output
image

### CODE REVIEW

File: artifacts/definitions/Windows/Forensics/RecycleBin.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+parameters:
+    - name: RecycleBinGlobs
+      default: C:\$Recycle.Bin\**\$I*
+
+precondition: SELECT OS From info() where OS = 'windows'
+
+sources:
+  - query: |
+        LET recyclebin <= SELECT * FROM foreach(
```
Comment (scudette) [inline artifacts/definitions/Windows/Forensics/RecycleBin.yaml:38]: There is probably no need to materialize the output into memory here - just select directly from foreach

File: vql/parsers/recyclebin.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+		if err != nil {
+			scope.Log("parse_recyclebin: %s", err)
+			return
+		}
+
+		for _, filename := range arg.Filenames {
+			func() {
+				defer utils.RecoverVQL(scope)
+				
+				err := vql_subsystem.CheckFilesystemAccess(scope, arg.Accessor)
```
Comment (scudette) [inline vql/parsers/recyclebin.go:71]: No need to do access checks again - you are doing it above.

File: vql/parsers/recyclebin/profile_gen.go [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+		utf[i/2] = o.Uint16(b[i:])
+	}
+	if len(b)/2 < len(utf) {
+		utf[len(utf)-1] = utf8.RuneError
+	}
+
+	return string(utf16.Decode(utf))
+}
+
+func ParseRecycleBin(reader io.ReaderAt) (*FileInfo, error) {
```
Comment (scudette) [inline vql/parsers/recyclebin/profile_gen.go:150]: Put your code in a different file so we can regenerate the strcuts if needed.

**Review by scudette — CHANGES_REQUESTED** [review]
Thanks! This is cool - just some minor changes.

File: vql/parsers/recyclebin/profile_gen.go (hunk 3 above)
Comment (svch0stz) [reply vql/parsers/recyclebin/profile_gen.go]: new file "parse_recycbin" with custom functions added

**Review by scudette — APPROVED** [review]

**Pull request merged**